core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala

```diff
@@ -145,7 +145,15 @@ private[storage] class BlockManagerDecommissioner(
                     // Confirm peer is not the fallback BM ID because fallbackStorage would already
                     // have been used in the try-block above so there's no point trying again
                     && peer != FallbackStorage.FALLBACK_BLOCK_MANAGER_ID) {
-                  fallbackStorage.foreach(_.copy(shuffleBlockInfo, bm))
+                  try {
+                    fallbackStorage.foreach(_.copy(shuffleBlockInfo, bm))
+                  } catch {
+                    case e: FileNotFoundException =>
+                      logWarning("Skipping block $shuffleBlockInfo, block deleted.", e)
+                    case NonFatal(e) =>
+                      logError(s"Fallback storage for $shuffleBlockInfo failed", e)
+                      keepRunning = false
+                  }
```

Comment:
Drop this ? The existing `NonFatal` block at the end does this currently.

Reply:
This is different from the existing `NonFatal` block because it will retry the failed blocks but the existing one is really a catch-all and leave some blocks not retried.

Reply:
It was not clear from the PR description that this behavior was being attempted.
+CC @dongjoon-hyun as you know this part more.

Reply:
There isn't a behavior change. If we remove the added `NonFatal` block, this [section](https://github.com/apache/spark/blob/2ab7aa87c25a0fd9eaa6047f02465130e5b22a18/core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala#L171-L185) won't get executed. This means there are shuffle blocks that never trigger `numMigratedShuffles.incrementAndGet()` and the decommissioner will loop forever because the `numMigratedShuffles` is always less than `migratingShuffles`.

Reply:
Is this true?
> If we remove the added NonFatal block, this [section](https://github.com/apache/spark/blob/2ab7aa87c25a0fd9eaa6047f02465130e5b22a18/core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala#L171-L185) won't get executed.

We have line 166, doesn't it?
https://github.com/apache/spark/blob/2ab7aa87c25a0fd9eaa6047f02465130e5b22a18/core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala#L166-L168

Do you think you can provide a test case as the evidence for your claim, @ukby1234 ?